Bearded.Utilities/Noise/INoiseMap.cs

```diff
@@ -0,0 +1,41 @@
+namespace Bearded.Utilities.Noise
+{
+    public interface INoiseMap
+    {
+        /// <summary>
+        /// Returns the value of the noise map at the given coordinates in the noise map.
+        /// This method only works for values within the [0, 1) x [0, 1) range only (upper bounds exclusive).
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns></returns>
+        public double ValueAt(double x, double y);
```

Comment:
I have never been a fan of using the indexer, because it - in my opinion - removes a bunch of context about what the indexer means. An indexer can be used to access a list, or a map entry, or a continuous value? Having an actual method name here I believe makes it much clearer what the accessor actually means.

I'm aware this is subjective, yet I don't see strong arguments for an indexer apart from succinctness, which I don't believe is necessarily an advantage. I'm happy to be convinced otherwise.